### PR TITLE
Added support for Ada Language Server

### PR DESCRIPTION
**Pull request by thindil**

### CODE REVIEW

File: ale_linters/ada/adalsp.vim [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+" Description: Support for Ada Language Server
+
+call ale#Set('ada_lsp_executable', 'ada_language_server')
+
+function! ale_linters#ada#adalsp#GetRootDirectory(buffer) abort
+    return fnamemodify(bufname(a:buffer), ':p:h')
+endfunction
+
+call ale#linter#Define('ada', {
+\   'name': 'adalsp',
```
Comment (w0rp) [inline ale_linters/ada/adalsp.vim]: Let's name this `adals`, and name the variable aboev `ada_adals_executable`. That will match some existing linter names like `dls` for D, and how the options are named for that.

File: ale_linters/ada/adals.vim [hunk 2]
```diff
@@ -0,0 +1,16 @@
+" Author: Bartek Jasicki http://github.com/thindil
+" Description: Support for Ada Language Server
+
+call ale#Set('ada_adals_executable', 'ada_language_server')
```
Comment (timlag1305) [inline ale_linters/ada/adals.vim]: I was looking into adding the Ada language server a while back and totally forgot about it until now. In any case, I added a couple more variables that the user could set:
`ada_ls_project` and `ada_ls_encoding` which correspond to "ada.projectFile" and "ada.defaultCharset" as listed here https://github.com/AdaCore/ada_language_server/#how-to-use-the-vscode-extension

I could give you the diff if you're interested.

Comment (thindil) [reply ale_linters/ada/adals.vim]: That will be awesome :) Sure, please send me a diff, I will try add it as soon as possible.

Comment (timlag1305) [reply ale_linters/ada/adals.vim]:
```suggestion
call ale#Set('ada_adals_executable', 'ada_language_server')
call ale#Set('ada_adals_project, 'default.gpr')
call ale#Set('ada_adals_encoding', 'utf-8')

function! ale_linters#ada#adals#GetInitializationOptions(buffer) abort
    return {
    \   'ada.projectFile': ale#Var(a:buffer, 'ada_adals_project'),
    \   'ada.defaultCharset': ale#Var(a:buffer, 'ada_adals_encoding')
    \}
endfunction
```

File: ale_linters/ada/adals.vim [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    return fnamemodify(bufname(a:buffer), ':p:h')
+endfunction
+
+call ale#linter#Define('ada', {
+\   'name': 'adals',
+\   'lsp': 'stdio',
+\   'executable': {b -> ale#Var(b, 'ada_adals_executable')},
+\   'command': '%e',
+\   'project_root': function('ale_linters#ada#adals#GetRootDirectory'),
+\})
```
Comment (timlag1305) [inline ale_linters/ada/adals.vim]:
```suggestion
\   'initialization_options': function('ale_linters#ada#adals#GetInitializationOptions')
\})
```

**Comment by timlag1305** [comment]
I added inline diffs. I can also give you a patch file later tonight @thindil if that's easier.

**Comment by thindil** [comment]
> I added inline diffs. I can also give you a patch file later tonight @thindil if that's easier.

@timlag1305 thank you very much. I just added them to my fork and I will test them soon. Should be everything ok.

**Comment by kalkin** [comment]
Bump?

**Comment by thindil** [comment]
@kalkin All requested changes are done thus now we just wait :)

**Comment by timlag1305** [comment]
Yeah unfortunately the original maintainer doesn't really have the time to handle all these PRs. They tried to get others to step up, but I'm not sure how effective that was. I'm sure the project would be receptive if someone was willing to step up. I think in the mean time, if I wanted a specific PR in my version of ale, I would fork the project and roll the PRs into my fork.

**Comment by thindil** [comment]
@timlag1305 true, plus as far I know, my fork should be synced with main repo, thus you can use it instead of official and pokes me about syncing it :)

**Comment by hsanson** [comment]
ALE migrated from Travis to Github Actions (see #3548) so it is necessary to rebase this PR from latest master branch to trigger the required checks.

```
# Add dense-analysis as remote upstream on your local fork. Not 
# needed if already added.
git remote add upstream https://github.com/dense-analysis/ale.git

# Sync your local master with upstream master
git checkout master
git fetch upstream master
git merge upstream/master

# Rebase the PR branch with master
git checkout my-branch-name
git rebase master   # Fix any conflicts you may have

# Force push the updated PR branch to origin to trigger the checks
git push -f origin my-branch-name
```

**Comment by thindil** [comment]
@hsanson Thank you for pointing me there. The local fork was rebased as requested. Also, requested changes were done (a long time ago). If something other is needed to do here, please tell :smile:

**Review by hsanson — APPROVED** [review]
Thanks for the patience. Looks good to me.

**Pull request merged**